### PR TITLE
swagger and quick filter

### PR DESCRIPTION
**Pull request by ethn**

### CODE REVIEW

File: card/mod/machines/lib/javascript/decko/filter.js.coffee [hunk 1]
```diff
@@ -1,73 +1,4 @@
...
-
-
-  # $('body').on 'click', '._record-filter', (e) ->
-  #   f = filterFor($("._filter-widget:visible"))
-  #   f.removeField("year")
-  #   data = $(this).data "filter"
-  #   # f.reset()
-  #   f.restrict data["key"], data["value"]
-
+# filter object that manages dynamic sorting and filtering
```
Comment (ethn) [inline card/mod/machines/lib/javascript/decko/filter.js.coffee]: file was getting long.  split up the filter object and all the events.

File: decko/lib/decko/swagger.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+module Decko
+  # hacky first go at swagger generation.
+  #
+  # In decko, it really just converts yaml to ruby and back again.
+  #
+  # But it's useful for generating decko swagger docs.
+  class Swagger
```
Comment (ethn) [inline decko/lib/decko/swagger.rb]: At this point, the only thing the ruby representation buys us is a mechanism to merge wikirate and decko yaml.  But it's a start...

File: card/lib/card/format/render.rb [hunk 3]
```diff
@@ -181,8 +181,6 @@ def current_view view
         @current_view = old_view
       end
 
-      private
-
```
Comment (ethn) [inline card/lib/card/format/render.rb]: this broke staging.

File: card/mod/search/set/abstract/03_filter/quick_filters.haml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    - quick_filter_list.each do |hash|
+      - link_hash = JSON(hash)
+      - filter_key = hash.keys.first
+      - value = hash.values.first
+      %a{ "data-filter": link_hash,
+          href: "#",
+          class: "quick-filter-by-#{filter_key}",
+          title: "Quick Filter" }
+        = mapped_icon_tag filter_key
+        // FIXME: mapped_icon_tag is WikiRate concept
```
Comment (ethn) [inline card/mod/search/set/abstract/03_filter/quick_filters.haml]: do we want to make mapped_icon_tags a decko thing?  Or should I just put a method here that we can override.  (Leaning towards the second option).

Comment (xithan) [reply card/mod/search/set/abstract/03_filter/quick_filters.haml]: probably not worth the effort to make it a decko thing

**Pull request merged**